test/throttle_SUITE.erl

```diff
@@ -0,0 +1,214 @@
+-module(throttle_SUITE).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-compile([export_all, nowarn_export_all]).
+
+-define(DEFAULT_NO_PROCESSES, 10).
+-define(DEFAULT_INTERVAL, 60000). %% one minute
+
+all() ->
+    [
+     {group, api}
+    ].
+
+groups() ->
+    [
+     {api, [parallel],
+      [
+       start,
+       rate_zero_is_not_accepted,
+       low_rate_gets_remapped,
+       low_interval_get_remapped,
+       start_and_stop,
+       change_rate,
+       change_rate_gradually,
+       send_and_wait,
+       run_with_interval_zero_limits_only_number_of_parallel_executions,
+       pause_and_resume,
+       get_state
+     ]}
+    ].
+
+init_per_suite(Config) ->
+    application:ensure_all_started(amoc),
+    amoc_cluster:set_master_node(node()),
+    TelemetryEvents = [[amoc, throttle, Event] || Event <- [init, rate, request, execute, process]],
+    telemetry_helpers:start(TelemetryEvents),
+    Config.
+
+end_per_suite(_) ->
+    application:stop(amoc),
+    telemetry_helpers:stop(),
+    ok.
+
+init_per_testcase(_, Config) ->
+    Config.
+
+end_per_testcase(_, _Config) ->
+    ok.
+
+%%-----------------------------------------------------------------------------------
+%% test cases
+%%-----------------------------------------------------------------------------------
+
+start(_) ->
+    %% Starts successfully
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100)),
+    %% If requested to start again, returns information accordingly
+    ?assertMatch({ok, already_started},
+                 amoc_throttle:start(?FUNCTION_NAME, 100)),
+    ?assertMatch({error, wrong_reconfiguration},
+                 amoc_throttle:start(?FUNCTION_NAME, 100, ?DEFAULT_INTERVAL + 1)),
+    ?assertMatch({error, wrong_no_of_procs},
+                 amoc_throttle:start(?FUNCTION_NAME, 100, ?DEFAULT_INTERVAL,
+                                     ?DEFAULT_NO_PROCESSES + 1)).
+
+rate_zero_is_not_accepted(_) ->
+    ?assertMatch({error, invalid_throttle}, amoc_throttle:start(?FUNCTION_NAME, 0, 100, 1)).
+
+low_rate_gets_remapped(_) ->
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 2, 100, 1)),
+    State = get_state_of_one_process(?FUNCTION_NAME),
+    ?assertMatch(#{name := ?FUNCTION_NAME,
+                   interval := 100,
+                   delay_between_executions := 50},
+                 State),
+    assert_telemetry_event([amoc, throttle, process], error, ?FUNCTION_NAME, 2, 100).
```

Comment:
`'error'` atom is supplied as a `Count` parameter, this doesn't seem correct.

Reply:
It is: https://hexdocs.pm/amoc/telemetry.html#throttle-process-internals

Reply:
I mean, that 'error' value for Count looks like a mistake at first glance. I would rather rename Count into a LogLevel.

Reply:
Renamed it to Measurement for consistency with the map.